migrations/kubernetes-agent/KubernetesAgent.UpgradeManager/KubernetesAgentUpgradeManager.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Dynamic;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Octopus.Versioning;
+
+namespace KubernetesAgent.UpgradeManager;
+
+public record File(string Name, byte[] Content);
+
+public record ScriptResult(int ExitCode, string Output);
+
+public delegate Task<ScriptResult> ScriptExecutor(string script, File[] files, CancellationToken cancellationToken);
+
+public interface IKubernetesAgentUpgradeManager
+{
+    Task UpgradeAgent(string helmRelease,
+        string @namespace,
+        IVersion currentVersion,
+        IVersion newVersion,
+        ScriptExecutor scriptExecutor,
+        CancellationToken cancellationToken);
+
+    Task<IVersion?> GetLatestVersion(ScriptExecutor scriptExecutor, CancellationToken cancellationToken);
+}
+
+public class KubernetesAgentUpgradeManager : IKubernetesAgentUpgradeManager
+{
+    const string KubernetesAgentOciRegistry = "oci://registry-1.docker.io/octopusdeploy/kubernetes-agent";
+    const string KubernetesAgentPreReleaseOciRegistry = "oci://docker.packages.octopushq.com/kubernetes-agent";
+
+    static readonly (int Version, bool isPreRelease) PackageVersion = (2, true);
```

Comment:
We need to source this from somewhere, not sure exactly how yet. We could potentially get the Versioning GHA to update the version in a specific consts file? Or we could get the version out of the assemblies when packaged as a nuget package.

I prefer the first option because I want to make sure I can run the test locally and it'll work correctly, relying on it being in a packed nuget package is not as nice in that respect.